helpers/compute_task_keyspace.rb

```diff
@@ -22,7 +22,16 @@ def getKeyspace(task)
   elsif task.hc_attackmode == 'maskmode'
 
     # build hashcat keyspace command
-    cmd = hashcatbinpath + ' -a 3 ' + task.hc_mask + ' --keyspace'
+    # cmd = hashcatbinpath + ' -a 3 ' + task.hc_mask + ' --keyspace'
+    keyspace = 0
+    task.hc_mask.to_s.each_char do |char|
+      keyspace *= 26 if char == 'u' || char == 'l'
+      keyspace *= 10 if char == 'd'
+      keyspace *= 32 if char == 's'
+      keyspace *= 42 if char == 'a'
+      keyspace *= 256 if char == 'b'
+    end
+    return keyspace.to_i
```

Comment:
@hlakhan

Hey guy,

Keyspace is not the total possibilities of job. Keyspace is necessary to [split a job 
](https://hashcat.net/wiki/doku.php?id=frequently_asked_questions#what_is_a_keyspace)

In addition, your result is always 0. So keep the previous code.